libchisel/src/fromwat.rs

```diff
@@ -0,0 +1,96 @@
+extern crate wabt;
+
+use super::{ModuleCreator, ModuleError};
+use crate::utils::*;
+use parity_wasm::elements::{deserialize_buffer, Module};
+use wabt::Wat2Wasm;
+
+/// Struct on which ModuleCreator is implemented.
+pub struct FromWat<'a> {
+    pub filename: &'a String,
```

Comment:
In my opinion we don't really need `&String`. It is more flexible with `&'a str` because `&String` would force it to be an owned string, whereas `String` already can be borrowed as a string slice.

Reply:
Also, it is a bit more idiomatic to expose getter methods rather than having public members. But it is not super important in this context.

Reply:
Well if the input to `new()` is not a `str` than this wont work.

Reply:
`String` can be borrowed as `&str`

Reply:
https://doc.rust-lang.org/std/string/struct.String.html#impl-AsRef%3Cstr%3E

Reply:
> Also, it is a bit more idiomatic to expose getter methods rather than having public members. But it is not super important in this context.

Actually it doesn't need to be `pub`.